frontend/src/app/shared/pipes/string-to-subpermission.pipe.ts

```diff
@@ -0,0 +1,20 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+import { allPermissions, SubPermissions } from "../models/permission.model";
+
+@Pipe({
+    name: "stringToSubpermission",
+    standalone: true,
+})
+export class StringToSubpermissionPipe implements PipeTransform {
+    transform(value: string, ...args: unknown[]): SubPermissions | null {
```

Comment:
Skoro `args` nigdzie nie używamy, to lepiej wywalić (przy okazji jeśli coś takiego zostało w innych pipeach, to też możesz usunąć).